server/src/test/java/org/opensearch/index/remote/RemoteRefreshSegmentPressureServiceTests.java

```diff
@@ -0,0 +1,179 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.remote;
+
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.concurrency.OpenSearchRejectedExecutionException;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.index.shard.IndexShard;
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.test.IndexSettingsModule;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.TestThreadPool;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.IntStream;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class RemoteRefreshSegmentPressureServiceTests extends OpenSearchTestCase {
+
+    private ClusterService clusterService;
+
+    private ThreadPool threadPool;
+
+    private ShardId shardId;
+
+    private RemoteRefreshSegmentPressureService pressureService;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        threadPool = new TestThreadPool("remote_refresh_segment_pressure_settings_test");
+        clusterService = new ClusterService(
+            Settings.EMPTY,
+            new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS),
+            threadPool
+        );
+        shardId = new ShardId("index", "uuid", 0);
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+        threadPool.shutdownNow();
+    }
+
+    public void testIsSegmentsUploadBackpressureEnabled() {
+        pressureService = new RemoteRefreshSegmentPressureService(clusterService, Settings.EMPTY);
+        assertFalse(pressureService.isSegmentsUploadBackpressureEnabled());
+
+        Settings newSettings = Settings.builder()
+            .put(RemoteRefreshSegmentPressureSettings.REMOTE_REFRESH_SEGMENT_PRESSURE_ENABLED.getKey(), "true")
+            .build();
+        clusterService.getClusterSettings().applySettings(newSettings);
+
+        assertTrue(pressureService.isSegmentsUploadBackpressureEnabled());
+    }
+
+    public void testAfterIndexShardCreatedForRemoteBackedIndex() {
+        Settings settings = Settings.builder().put(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, "true").build();
+        IndexSettings indexSettings = IndexSettingsModule.newIndexSettings("test_index", settings);
+        ShardId testShardId = new ShardId("index", "uuid", 0);
+        IndexShard indexShard = mock(IndexShard.class);
+        when(indexShard.indexSettings()).thenReturn(indexSettings);
+        when(indexShard.shardId()).thenReturn(testShardId);
```

Comment:
Can this be refactored to a helper method? It looks like maybe only the `SETTING_REMOTE_STORE_ENABLED` boolean is the only value that varies here.